test/framework/util.go

```diff
@@ -77,8 +77,8 @@ func ReadFromURI(uri string) (string, error) {
 	return string(data), nil
 }
 
-// WaitFor waits for a specific condition to be met
-func WaitFor(namespace, display string, timeout time.Duration, condition func() (bool, error)) error {
+// WaitFor waits for a specification condition to be met or until one error condition is met
+func WaitFor(namespace, display string, timeout time.Duration, condition func() (bool, error), errorConditions []func() (bool, error)) error {
```

Comment:
why don't we have a varargs as well here for `errorConditions` ?

Reply:
This was the first thing I did, but golang does not support varargs when this is sent to different methods:

1.- errorConditions as varargs in WaitForOnOpenshift
2.- In WaitForOnOpenshift, send the errorConditions to WaitFor which is also set as varargs -> Compiler error because golang things errorConditions is an array, so it does not match with the varargs contract (I know, this with Java does not happen)

Therefore, we would need two WaitFor methods, one using an array of errorCondition and another using errorCondition as varargs. But taking into account we don't use the WaitFor directly anywhere (at least now), having the second method is useless.

Reply:
https://github.com/kiegroup/kogito-cloud-operator/blob/8f96a5c892f619967472e9021d62bfaaa2459c90/test/framework/cli.go#L52

Reply:
taking varargs and giving varargs :)

Reply:
haha I see dots, lot of dots.
Thanks Tristan, will update this and I just learned something new! :)

Reply:
Changed.